### ISSUE TITLE
Parse Cmdline and Exception in windbg

### ISSUE THREAD

**Issue by Saturn35** [0vercl0k/wtf#39]
Firstly，A great work you have done!
But when I started to try my own dump and regs, there were some problems.

dump and json file are created(1 vitrualprocessor & 1G mem in windbg kd mode)。
![image](https://user-images.githubusercontent.com/53106646/140319312-67ec00ec-90f1-488d-9578-2b082c6596ce.png)

and my fuzzer module is just like the fuzzer_hevd.cc module，then use build-release-msvc.bat in VS2019 Developer Command Promp，build successfully.

prepared 4 dirs:
![image](https://user-images.githubusercontent.com/53106646/140319999-f22c2927-1734-4f20-b4c0-c25ea19353d3.png)

Server side run normally,but when i run client side like my bat script

![image](https://user-images.githubusercontent.com/53106646/140320546-e883e66a-696d-481b-a9e5-3ebf63c80618.png)

,it crashed with stop work pop-ups，then debug with windbg，it seems that parse cmdline has some problems。
![image](https://user-images.githubusercontent.com/53106646/140321426-f7c27459-4d50-44f8-bd71-0604a55be002.png)
`0:000> g
(312c.2420): C++ EH exception - code e06d7363 (first chance)
ModLoad: 00007ff8`62c00000 00007ff8`62c12000   C:\WINDOWS\SYSTEM32\kernel.appcore.dll
ntdll!NtTerminateProcess+0x14:
00007ff8`66ccd2f4 c3              ret`

Then,I Try the HEVD ,I found that even its client fuzz process run normally，it also can't debug wiht windbg.
What might be wrong with me? (1, are all executed to parse the command line crash, 2, windbg can't see the cause of the crash?)

environment: HEVD with commit 1b08f6691cd949b39f65332e3f82782f795ee4bb
                     TestFuzzer with commit 16ebc8ba5972841f0e83e0c62098602040db77d3

**Comment by 0vercl0k** [comment]
Thanks for the report! I'll have a look at this this week-end :)

Cheers

**Comment by Saturn35** [comment]
OK, thank you for your enthusiastic and quick reply, add some details below

I open the event settings in windbg,open  break with C++ EH exceptions.
got stack trace like this:

test fuzzer in commit 16ebc8ba5972841f0e83e0c62098602040db77d3
![image](https://user-images.githubusercontent.com/53106646/140453764-c99b193f-255f-427c-86ca-ef4785a2ffb1.png)


HEVD in commit 1b08f6691cd949b39f65332e3f82782f795ee4bb
![image](https://user-images.githubusercontent.com/53106646/140453829-0e92647b-c146-4a49-a78e-0d90dc8d09f6.png)
but when I run fuzz node from cmd,it looks work normally...
![image](https://user-images.githubusercontent.com/53106646/140453935-437ab63d-42d2-4bd6-b90f-6ab6c6b5b1b7.png)

It seems that this **wtf!CLI::Option::run_callback** func which I didn't get it.(not sure if this is the reason why my module can't run)

maybe I forget something in debugging,thx for your reply again!

**Comment by Saturn35** [comment]
my pop-ups crash I got it! 

This is due to the problem of the tsc_aux msx register in regs.json file. Because the dump of the tsc_aux register status was temporarily shielded by using bdump.js before, the subsequent wtf process parsed the register status error. Since my Hyper-V VM is nested in VMware, I am not very clear about what happened.

Why i blocked it is because that script uses **rdmsr 0xc0000103** command, display **no such msr**, I think it may be similar to the protection caused by the Virtualization Based Security mechanism? It has not been resolved yet, my temporary solution is

1) Change to a physical machine + Hyper-V VM, it seems that the above problem does not exist.
2) Since the status of the tsc register I dumped is 0, the low 32-bit tsc_aux is also manually set to 0, which seems to be feasible at present.

Ok, the next step is about the C++ EH exception encountered in windbg debugging:)
